data-serving/reusable-data-service/data_service/main.py

```diff
@@ -150,8 +152,18 @@ def excluded_case_ids():
         return jsonify({"message": e.args[0]}), e.http_code
 
 
+@app.route("/api/schema", methods=["POST"])
+def add_field_to_case_schema():
+    try:
+        req = request.get_json()
+        schema_controller.add_field(req["name"], req["type"], req["description"])
+        return "", 201
```

Comment:
`except ConflictError`  returns 409, `except PreconditionUnsatisfiedError` returns 400,
and `except KeyError` returns 400?

Reply:
so `KeyError` shouldn't happen: that will come from `Field.model_type` if you try to use a field type that isn't in the allowed list. So we turn it into `PreconditionUnsatisfiedError` which is the same as when we explicitly check that the type isn't found in the allowed list: i.e. both exception paths mean the same thing so they result in the same error (400 bad request: you've asked for something we aren't going to let you do).

`ConflictError` does indeed result in a 409 Conflict status.